test/wallet-records-test.js

```diff
@@ -0,0 +1,425 @@
+      const {data, block, tx} = getRandomTXRecordData(true, true);
+      const wtx = new TXRecord(tx, block);
+
+      compareTXRecord(wtx, data);
+    });
+
+    it('should fail encode w/o tx', () => {
+      let err;
+      try {
+        const wtx = new TXRecord();
+        wtx.encode();
+      } catch (e) {
+        err = e;
+      }
+
+      assert(err, 'Should throw error w/o tx');
+    });
+
+    it('should encode/decode w/ tx', () => {
+      const {data, tx} = getRandomTXRecordData(true);
+
+      const wtx = new TXRecord(tx);
+      const encoded = wtx.encode();
+      const decoded = TXRecord.decode(encoded);
+
+      compareTXRecord(wtx, data);
+      compareTXRecord(decoded, data);
+    });
+
+    it('should encode/decode w/ tx and block', () => {
+      const {data, tx, block} = getRandomTXRecordData(true, true);
+
+      const wtx = new TXRecord(tx, block);
+      const encoded = wtx.encode();
+      const decoded = TXRecord.decode(encoded);
+
+      compareTXRecord(wtx, data);
+      compareTXRecord(decoded, data);
+    });
+
+    it('should initialize from TX', () => {
+      const {data, tx} = getRandomTXRecordData(true);
+      const wtx = TXRecord.fromTX(tx);
+      compareTXRecord(wtx, data);
+    });
+
+    it('should initialize from TX and Block', () => {
+      const {data, tx, block} = getRandomTXRecordData(true, true);
+      const wtx = TXRecord.fromTX(tx, block);
+      compareTXRecord(wtx, data);
+    });
+
+    it('should set and unset block', () => {
+      const {data, tx, block} = getRandomTXRecordData(true, true);
+      const wtx = TXRecord.fromTX(tx);
+
+      assert.strictEqual(wtx.getBlock(), null);
+      assert.strictEqual(wtx.getDepth(random.randomInt()), 0);
+      compareTXRecord(wtx, {
+        ...data,
+        ...emptyBlock
+      });
+
+      wtx.setBlock(block);
+      assert.strictEqual(wtx.getDepth(0), 0);
+      assert.strictEqual(wtx.getDepth(block.height), 1);
+      assert.strictEqual(wtx.getDepth(block.height + 1000), 1001);
+      compareBlockMeta(wtx.getBlock(), block);
+      compareTXRecord(wtx, data);
+
+      wtx.unsetBlock(block);
+      assert.strictEqual(wtx.getBlock(), null);
+      assert.strictEqual(wtx.getDepth(random.randomInt()), 0);
+      compareTXRecord(wtx, {
+        ...data,
+        ...emptyBlock
+      });
+    });
+  });
+
+  describe('MapRecord', function() {
+    it('should initialize with default', () => {
+      const map = new MapRecord();
+
+      assert.strictEqual(map.wids.size, 0);
+    });
+
+    it('should encode/decode empty map', () => {
+      const map = new MapRecord();
+      const encoded = map.encode();
+      const decoded = MapRecord.decode(encoded);
+
+      assert.bufferEqual(encoded, Buffer.from('00'.repeat(4), 'hex'));
+      assert.strictEqual(map.wids.size, 0);
+      assert.strictEqual(decoded.wids.size, 0);
+    });
+
+    it('should encode/decode map', () => {
+      const rand = random.randomRange(1, 100);
```

Comment:
Same here, why not `const` an array of 10 hard coded ints? I reccomend [The Numbers](https://lostpedia.fandom.com/wiki/The_Numbers) :-P

Reply:
Here it does not really matter whether it's fixed or not lol. Still went with random number of entries in the map.
Next test is using fixed value of `20`.

Reply:
okie doke.